### PR TITLE
Added 4 files with JUnit tests.

### PR DESCRIPTION
**Pull request by dmurasov**
Unit tests were written for the BasicValidationService, InMemUserPerository, InMemPaymentRepository, PaymentService classes. Mockito was used for the PaymentService test.

### CODE REVIEW

File: src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+public class InMemPaymentRepositoryTest {
+
+    private final InMemPaymentRepository inMemPaymentRepository = new InMemPaymentRepository();
+
+    private final Payment payment1 = new Payment(1, 100.0, "Payment 1");
+
+    private final Payment payment2 = new Payment(2, 20.0, "Payment 2");
+
+    @Test
+    void findById_shouldThrowAnExceptionIfIDisNull() {
```
Comment (perepiolcamaxim) [inline src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java]: uppercase ...IsNull()

File: src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    @Test
+    void findByID_ShouldReturnEmptyOptionalIfItIsNotValidID() {
+        Optional<Payment> current = inMemPaymentRepository.findById(payment1.getPaymentId());
+        assertEquals(Optional.empty(), current);
+    }
+
+    @Test
+    void findByID_ShouldReturnPaymentIfIDisValid() {
```
Comment (perepiolcamaxim) [inline src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java]: same here

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java [hunk 3]
```diff
@@ -1,31 +1,118 @@
...
+    private final static User user = new User(id, "Dmitry", Status.ACTIVE);
+
+    private final static Payment payment = new Payment(user.getId(), amount, "Payment from user " + user.getName());
+
+    private final static String message = "New Message";
+
+    private final static UUID uuid = UUID.randomUUID();
+
+    @BeforeEach
+    void setUp() {}
```
Comment (perepiolcamaxim) [inline src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java]: remove if it is empty

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java [hunk 4]
```diff
@@ -1,31 +1,118 @@
...
 
     @Test
     void createPayment() {
+        when(userRepository.findById(anyInt())).thenReturn(Optional.of(user));
```
Comment (perepiolcamaxim) [inline src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java]: please avoid using anyInt(), any() etc. If it is possible use concrete values

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java [hunk 5]
```diff
@@ -1,31 +1,118 @@
...
+
+    private final static Integer id = 1;
+
+    private final static User user = new User(id, "Dmitry", Status.ACTIVE);
+
+    private final static Payment payment = new Payment(user.getId(), amount, "Payment from user " + user.getName());
+
+    private final static String message = "New Message";
+
+    private final static UUID uuid = UUID.randomUUID();
```
Comment (perepiolcamaxim) [inline src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java]: if object is used only locally no need to make it a constant, and also constants should be in uppercase, AMOUNT, ID, USER ...

**Review by perepiolcamaxim — APPROVED** [review]

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java (hunk 5 above)
Comment (dmurasov) [reply src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java]: changed

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java (hunk 4 above)
Comment (dmurasov) [reply src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java]: changed

File: src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java (hunk 3 above)
Comment (dmurasov) [reply src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java]: method was removed

File: src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java (hunk 2 above)
Comment (dmurasov) [reply src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java]: changed

File: src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java (hunk 1 above)
Comment (dmurasov) [reply src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java]: changed

**Pull request merged**